### PR TITLE
Add `additional_zip_keys` for migrations

### PR DESCRIPTION
**Pull request by 0xbe7a**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by xhochy** [comment]
@0xbe7a Can you link a PR showing that this does produce the wanted output, e.g. based on my PR https://github.com/conda-forge/pydantic-core-feedstock/pull/63 ?

**Comment by 0xbe7a** [comment]
See [conda-forge/pydantic-core-feedstock@`3ff74ee` (#63)](https://github.com/conda-forge/pydantic-core-feedstock/pull/63/commits/3ff74eee261f4b8b2ab0626c9026966f6a51e35b)

and the subsequent rerender [conda-forge/pydantic-core-feedstock@`e2db1a6` (#63)](https://github.com/conda-forge/pydantic-core-feedstock/pull/63/commits/e2db1a6da0c6df26af143bb4a4e415d1b478c53e)

**Review by xhochy — APPROVED** [review]
Thanks @0xbe7a

**Comment by isuruf** [comment]
This would break for a migration that comes after with the same primary key.

**Comment by isuruf** [comment]
You need to keep the additional key (`channel_sources`) and the value to be duplicated (`conda-forge`) and re-use that for all migrations that comes after.

File: conda_smithy/variant_algebra.py [hunk 1]
```diff
@@ -126,15 +126,41 @@ def op_variant_key_add(v1: dict, v2: dict):
...
+                        newly_added_zip_keys.add(additional_key)
+                break
+        else:
+            # The for loop didn't break thus the primary is not part of any zip_key,
+            # create a new one including the primary key
+            result.setdefault("zip_keys", []).append(
+                [primary_key] + additional_zip_keys
+            )
+            newly_added_zip_keys.update([primary_key] + additional_zip_keys)
+
```
Comment (isuruf) [inline conda_smithy/variant_algebra.py:157]: This should be moved to after line 162 as this should only be used for key_add operation.

Comment (0xbe7a) [reply conda_smithy/variant_algebra.py:157]: Doesn't `op_variant_key_add` only get used for key_add operations? 😅

**Comment by 0xbe7a** [comment]
Good catch, i opened #1769

File: conda_smithy/variant_algebra.py (hunk 1 above)
Comment (isuruf) [reply conda_smithy/variant_algebra.py:157]: It should, but I'm not sure why Line 160 returns fast.

Comment (0xbe7a) [reply conda_smithy/variant_algebra.py:157]: I see

**Pull request merged**